language/stdlib/src/lib.rs

```diff
@@ -9,33 +9,94 @@ use bytecode_verifier::{batch_verify_modules, VerifiedModule};
 use move_lang::{move_compile, shared::Address, to_bytecode::translate::CompiledUnit};
 use once_cell::sync::Lazy;
 use std::path::PathBuf;
+use vm::file_format::CompiledModule;
 
 pub const STD_LIB_DIR: &str = "modules";
 pub const MOVE_EXTENSION: &str = "move";
 
-static MOVELANG_STDLIB: Lazy<Vec<VerifiedModule>> = Lazy::new(build_stdlib);
+pub const NO_USE_STAGED: &str = "MOVE_NO_USE_STAGED";
+
+pub const TRANSACTION_SCRIPTS: &str = "transaction_scripts";
+/// The output path under which staged files will be put
+pub const STAGED_OUTPUT_PATH: &str = "staged";
+/// The file name for the staged stdlib
+pub const STAGED_STDLIB_NAME: &str = "stdlib";
+/// The extension for staged files
+pub const STAGED_EXTENSION: &str = "mv";
+
+// The current stdlib that is freshly built
+static FRESH_MOVELANG_STDLIB: Lazy<Vec<VerifiedModule>> = Lazy::new(build_stdlib);
+
+// The staged version of the move standard library.
+static STAGED_MOVELANG_STDLIB: Lazy<Vec<VerifiedModule>> = Lazy::new(|| {
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push(STAGED_OUTPUT_PATH);
+    path.push(STAGED_STDLIB_NAME);
+    path.set_extension(STAGED_EXTENSION);
+    let buf = std::fs::read(path).unwrap();
+    let modules = lcs::from_bytes::<Vec<Vec<u8>>>(&buf)
+        .unwrap()
+        .into_iter()
+        .map(|bytes| CompiledModule::deserialize(&bytes).unwrap())
+        .collect();
+    batch_verify_modules(modules)
+});
+
+/// An enum specifying whether the staged stdlib/scripts should be used or freshly built versions
```

Comment:
It might be a good idea to elaborate on what "staged" means in this comment, including the basic motivation to have this.